### PR TITLE
feat : add memory pack

### PR DESCRIPTION
**Pull request by AminMatini**
How to make the fastest .NET Serializer with .NET 7 / C# 11, case of MemoryPack

### CODE REVIEW

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: Please remove ImplicitUsings line.

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <Nullable>enable</Nullable>
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: Please remove Nullable line.

File: src/CSharp/EasyMicroservice.MemoryPack/Sample.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿using MemoryPack;
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservice.MemoryPack/Sample.cs]: Please add your sample file to the test cases.

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: Rename the "EasyMicroservice" folder's name to "EasyMicroservices".
However please fix the namespaces.
Please move "EasyMicroservice.MemoryPack" to "EasyMicroservices.Serialization.MemoryPack"
Please create a Providers folder and add MemoryPack Provider to it.
you can check other projects like EasyMicroservices.Serialization.System.Text.Json and EasyMicroservices.Serialization.BinaryGo

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]
Please check the request changes.

**Review by mahdidelzendeh72 — COMMENTED** [review]
please follow project Structure

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj (hunk 1 above)
Comment (AminMatini) [reply src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: thank you for attention  . 
conversion solved

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj (hunk 2 above)
Comment (AminMatini) [reply src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: conversion solved

File: src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj (hunk 4 above)
Comment (AminMatini) [reply src/CSharp/EasyMicroservice.MemoryPack/EasyMicroservice.MemoryPack.csproj]: conversion solved

File: src/CSharp/EasyMicroservices.Serialization.MemoryPack/EasyMicroservices.Serialization.MemoryPack.csproj [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.Serialization.MemoryPack/EasyMicroservices.Serialization.MemoryPack.csproj:4]: does it not support another dotnet version? if it supports it, then please add multiple target frameworks.

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]
Please check the request change @AminMatini. Thanks.

File: src/CSharp/EasyMicroservices.Serialization.MemoryPack/EasyMicroservices.Serialization.MemoryPack.csproj (hunk 5 above)
Comment (AminMatini) [reply src/CSharp/EasyMicroservices.Serialization.MemoryPack/EasyMicroservices.Serialization.MemoryPack.csproj:4]: other versions are also supported, but according to the document of this package, it is better to use this package in version 7.NET

File: src/CSharp/EasyMicroservice.MemoryPack/Sample.cs (hunk 3 above)
Comment (Ali-YousefiTelori) [reply src/CSharp/EasyMicroservice.MemoryPack/Sample.cs]: @AminMatini can you please add this file to the test cases and add test cases for this project?

**Review by Ali-YousefiTelori — APPROVED** [review]
@AminMatini Thanks for your contributions.

**Review by mahdidelzendeh72 — APPROVED** [review]

**Pull request merged**